pyvista/examples/downloads.py

```diff
@@ -3,311 +3,240 @@
-    """Verify that the version of the cache matches the expected version.
-
-    Clears cache when there is a version mismatch. This avoids any potential
-    issues with old file structures due to changed download methods.
-
-    """
-    cache_version_file = os.path.join(pyvista.EXAMPLES_PATH, 'VERSION')
-    cache_version = _cache_version(cache_version_file)
-
-    # clear with no version file or an old cache version
-    if cache_version < CACHE_VERSION:
-        delete_downloads()
-        with open(cache_version_file, 'w') as fid:
-            fid.write(str(CACHE_VERSION))
-
-
-def delete_downloads():
-    """Delete all downloaded examples to free space or update the files.
-
-    Returns
-    -------
-    bool
-        Returns ``True``.
-
-    Examples
-    --------
-    Delete all local downloads.
+# disable pooch verbose logging
+POOCH_LOGGER = get_logger()
+POOCH_LOGGER.setLevel(logging.CRITICAL)
 
-    >>> from pyvista import examples
-    >>> examples.delete_downloads()  # doctest:+SKIP
-    True
-
-    """
-    _check_examples_path()
-    if os.path.isdir(pyvista.EXAMPLES_PATH):
-        shutil.rmtree(pyvista.EXAMPLES_PATH)
-    os.makedirs(pyvista.EXAMPLES_PATH)
-    return True
 
+CACHE_VERSION = 3
 
-def _decompress(filename, output_path):
-    _check_examples_path()
-    zip_ref = zipfile.ZipFile(filename, 'r')
-    zip_ref.extractall(output_path)
-    return zip_ref.close()
+# If available, a local vtk-data instance will be used for examples
+if 'VTK_DATA_PATH' in os.environ:  # pragma: no cover
+    _path = os.environ['VTK_DATA_PATH']
 
+    if not os.path.basename(_path) == 'Data':
+        # append 'Data' if user does not provide it
+        _path = os.path.join(_path, 'Data')
+    if not os.path.isdir(_path):
+        warnings.warn(f"VTK_DATA_PATH: {_path} is an invalid path and will be ignored")
+    else:
+        # pooch assumes this is a URL so we have to take care of this
+        if not _path.endswith('/'):
+            _path = _path + '/'
+        SOURCE = _path
+        _FILE_CACHE = True
+
+else:
+    SOURCE = "https://github.com/pyvista/vtk-data/raw/master/Data/"
+    _FILE_CACHE = False
+
+# allow user to override the local path
+if 'PYVISTA_USERDATA_PATH' in os.environ:  # pragma: no cover
+    if not os.path.isdir(os.environ['PYVISTA_USERDATA_PATH']):
+        warnings.warn('Ignoring invalid {PYVISTA_USERDATA_PATH')
+    else:
+        PATH = os.environ['PYVISTA_USERDATA_PATH']
+else:
+    # use default pooch path
+    PATH = str(pooch.os_cache(f'pyvista_{CACHE_VERSION}'))
 
-def _get_vtk_file_url(filename):
-    return f'https://github.com/pyvista/vtk-data/raw/master/Data/{filename}'
+    # provide helpful message if pooch path is inaccessible
+    if not os.path.isdir(PATH):  # pragma: no cover
+        try:
+            os.makedirs(PATH, exist_ok=True)
+            if not os.access(PATH, os.W_OK):
+                raise OSError
+        except (PermissionError, OSError):
+            warnings.warn(
```

Comment:
Normally I'd say this would be a best practice, but in practice this sometimes causes anything depending on PyVista's example module to fail on import because of permission errors.

Regardless I'm adding a note.

Reply:
I forgot that we have non downloaded data in this module too.   Ignore these comments about converting to errors.